staging/src/kubevirt.io/client-go/generated/kubevirt/clientset/versioned/typed/core/v1/fake/fake_virtualmachine_expansion.go

```diff
@@ -0,0 +1,120 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors
+ *
+ */
+
+package fake
+
+import (
+	"context"
+
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/testing"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+	fake2 "kubevirt.io/client-go/kubecli/fake"
+)
+
+func (c *FakeVirtualMachines) GetWithExpandedSpec(ctx context.Context, name string) (*v1.VirtualMachine, error) {
+	obj, err := c.Fake.
+		Invokes(testing.NewGetSubresourceAction(virtualmachinesResource, c.ns, "expand-spec", name), &v1.VirtualMachine{})
+
+	if obj == nil {
+		return nil, err
+	}
+	return obj.(*v1.VirtualMachine), err
+}
+
+func (c *FakeVirtualMachines) PatchStatus(ctx context.Context, name string, pt types.PatchType, data []byte, patchOptions k8smetav1.PatchOptions) (result *v1.VirtualMachine, err error) {
+	return c.Patch(ctx, name, pt, data, patchOptions, "status")
+}
+
+func (c *FakeVirtualMachines) Restart(ctx context.Context, name string, restartOptions *v1.RestartOptions) error {
+	_, err := c.Fake.
+		Invokes(fake2.NewPutSubresourceAction(virtualmachinesResource, c.ns, "restart", name), nil)
```

Comment:
I think it would be great if we can have a new struct, let's call it `restartActionImpl` that would also provide the options. Then a reactor could access these as well.